### ISSUE TITLE
Make colour in BuildPackages.sh logs optional (or remove?)

### ISSUE THREAD

**Issue by alex-konovalov** [gap-system/gap#1120]
After #1023 by @hungaborhorvath, `BuildPackages.sh` uses coloured output. This may be helpful for interactive usage, but not for viewing Jenkins logs - that's what I see there:

```
Sun  5 Feb 00:29:47 GMT 2017

[32m==== Checking radiroot[0m
[32mNo building required for radiroot[0m

Sun  5 Feb 00:29:47 GMT 2017

[32m==== Checking rcwa-4.4.1[0m
[32mNo building required for rcwa-4.4.1[0m

Sun  5 Feb 00:29:47 GMT 2017

[32m==== Checking rds[0m
[32mNo building required for rds[0m
```

Is it possible to make it optional? For me, being able to call it with e.g.
```
../bin/BuildPackages.sh --no-colour
``` 
would work.

**Comment by hungaborhorvath** [comment]
What I can do more or less quickly is to add an option to remove the colouring _everywhere_. 

The good fix would be, though, to colour the output on the screen but not in the logs. I am not entirely sure if that is easily doable.

**Comment by ChrisJefferson** [comment]
This is an area where it is hard to be perfect.

A patch is (hopefully) on it's way -- this will detect if you are writing to a file rather than to a terminal output.

**Comment by alex-konovalov** [comment]
Thanks - may not work automatically for Jenkins, which logs all terminal output, but indeed may be helped by a shell variable or tweaking the Jenkins job script. Will look at PR.

**Comment by hungaborhorvath** [comment]
@alex-konovalov @ChrisJefferson I tried another way to do this properly in #1122, because it seems to me that #1121 would not work. However, it now needs to ````sleep```` at some points due to a bug I am unable to debug. Anyone could check what the problem might be there? Thanks.

**Comment by ChrisJefferson** [comment]
@hungaborhorvath : In short, I think you are being too clever :) Just don't print out the colors, rather than try to strip them back out later.

**Comment by hungaborhorvath** [comment]
@ChrisJefferson I somehow do not see how that would work. Currently we have coloured output, and then the output is redirected into files using ````tee````. That is, the output as a whole is redirected. So I do not see how we could colour on terminal and not in files in any other way than to strip off the colouring when doing the logging.

Or do you mean get rid of the colours completely, even on terminal?

**Comment by hungaborhorvath** [comment]
Very well, #1123 adds the option to turn of colouring completely. That can be merged quickly, and fixes this issue. In the meantime, we can discuss what went wrong with #1122 to only colour to terminal, and never not to log files.

**Comment by ChrisJefferson** [comment]
Sorry for not explaning myself well.

I meant something like my patch, but with `--no-color` instead of trying to guess if we are on a terminal.

**Comment by hungaborhorvath** [comment]
@ChrisJefferson Something like #1123?

**Comment by fingolfin** [comment]
I think we need both -- the --no-color flag *and* turning off colors when output of this script is piped into a file, which is what @alex-konovalov's scripts do. Hence I included both in #1128. This way, hopefully Alex doesn't have to make any changes to his scripts.

**Comment by fingolfin** [comment]
Fixed by #1128.
